### PR TITLE
Implement NotEqualClauseInQuery.

### PR DESCRIPTION
**Pull request by dairyo**
I implemented `NotEqualClauseInQuery` and its tests.

Related PR: #303

### CODE REVIEW

File: ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift [hunk 1]
```diff
@@ -12,6 +12,80 @@ import XCTest
 
 extension XCTestCase {
 
+    func ThingIFTestEqualAny(
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift]: Should function name start with small letter?  like `equalsAny`

File: ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift [hunk 2]
```diff
@@ -12,6 +12,80 @@ import XCTest
...
+            return
+        } else {
+            let errorMessage = message ?? "Types are different."
+            XCTFail(
+              "file=\(file), line=\(line): \(errorMessage)")
+            return
+        }
+    }
+
+    func ThingIFTestEqualDictionary(
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift]: Should function name start with small letter? like `equalsDic`

File: ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift [hunk 3]
```diff
@@ -12,6 +12,80 @@ import XCTest
...
 extension XCTestCase {
 
+    func ThingIFTestEqualAny(
+      _ expected:  Any?,
+      _ actual: Any?,
+      _ message: String? = nil,
+      _ file: StaticString = #file,
+      _ line: UInt = #line)
+    {
+        if expected == nil && expected == nil {
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift]: `if expected == nil && actual == nil`

File: ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift [hunk 4]
```diff
@@ -12,6 +12,80 @@ import XCTest
...
+    }
+
+    func ThingIFTestEqualDictionary(
+      _ expected: [String : Any]?,
+      _ actual: [String : Any]?,
+      _ message: String? = nil,
+      _ file: StaticString = #file,
+      _ line: UInt = #line)
+    {
+        if expected == nil && expected == nil {
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDKTests/XCTest+Utils.swift]: `if expected == nil && actual == nil`

**Comment by dairyo** [comment]
@yongpingchen I fixed according to your comment.

I rename `ThingIFTestEqualXXX` methods to `asssertEqualsXXX`.

**Comment by yongpingchen** [comment]
LGTM

**Pull request merged**